backend/internal/pkg/services/car/car_test.go

```diff
@@ -0,0 +1,82 @@
+package car
+
+import (
+	"context"
+	"testing"
+
+	// "github.com/ParkWithEase/parkeasy/backend/internal/pkg/models"
+	"github.com/jackc/pgconn"
+	"github.com/stretchr/testify/assert"
+    // "github.com/jackc/pgx/v4"
+)
+
+// // TestGetCarsByUserID tests GetCarsByUserID function
+// func TestGetCarsByUserID(t *testing.T) {
+// 	mockDB := &MockDB{
```

Comment:
My recommendation is to put another layer of abstraction onto this DB, and make a database interface (we call it repo for some reason in our code base) that return your Car entry directly instead of pgx.Rows. and use that for mocking. Then when implementing, you can directly do the query in your implementation instead of having a DB interface.
Have a look at [repositories ](https://github.com/ParkWithEase/parkeasy/tree/main/backend/internal/pkg/repositories) for inspiration. We have database interface class (with GET/ DELETE/ ... method) and we use those to implement an in memory kinda database. You can absolutely do that using Postgres implementation instead. The test can be mock using that interface instead.